File: implementations/node/sdk/src/contracts/control/execution-input-submission-request.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkJsonObject } from "../../json/json-value.js";

export interface AiSdkExecutionInputSubmissionRequest {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.executionInputSubmissionRequest;
  readonly waitingKey: string;
  readonly waitingStepName?: string;
  readonly input: AiSdkJsonObject;
  readonly reason?: string;
  readonly correlationId?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/control/execution_input_submission_request.py
from dataclasses import dataclass, field

from ...json_types import AiSdkJsonObject, AiSdkJsonValue
from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions


@dataclass(frozen=True)
class AiSdkExecutionInputSubmissionRequest(AiSdkWireModel):
    waiting_key: str
    input: AiSdkJsonObject = field(metadata={"json": True})
    schema_version: int = AiSdkSchemaVersions.EXECUTION_INPUT_SUBMISSION_REQUEST
    waiting_step_name: str | None = None
    reason: str | None = None
    correlation_id: str | None = None
